### ISSUE TITLE
lightning-cli connect to WRONG IP misrepresents success

### ISSUE THREAD

**Issue by dscotese** [ElementsProject/lightning#4185]
### Issue and Steps to Reproduce
I misused c-clightning to test whether or not my laptop node is connected with `lightning-cli connect ID@IP:port` from a different machine on a different LAN.  It reports success no matter what IP address I use.  This suggests to me that "connect" is broken, and probably because it's using cached information to "answer the instruction to connect" which is very misleading.

### `getinfo` output
```
{ 
   "id": "03e1b9e6ca5d912254060927794efd3b96a8149f9d360bbec68e693fe7c47a8235",
   "alias": "MemeracingLN",
   "color": "68f442",
   "num_peers": 17,
   "num_pending_channels": 0,
   "num_active_channels": 17,
   "num_inactive_channels": 0,
   "address": [
      {
         "type": "ipv4",
         "address": "172.81.183.236",
         "port": 9735
      },
      {
         "type": "torv3",
         "address": "7porym5sb6d6yxil4egvxshzuy2vjgwxhdaei2ri2xvlodtzdcwyolid.onion",
         "port": 9735
      }
   ],
   "binding": [
      {
         "type": "ipv4",
         "address": "0.0.0.0",
         "port": 9735
      }
   ],
   "version": "0.9.0-1",
   "blockheight": 655688,
   "network": "bitcoin",
   "msatoshi_fees_collected": 363316,
   "fees_collected_msat": "363316msat",
   "lightning-dir": "/home/ubuntu/.lightning/bitcoin"
}
```
## Expected Behavior ##
If the node is already connected, output should be prefaced with:
`Already connected to [URI to which it's connected, ID@IP:PORT]:`
This would make it clear when the user has requested to connect on one IP:PORT but c-lightning is NOT doing that but still reporting whatever it reports.

## Current Behavior ##
It currently reports that it is connected without any indication of the IP or port through which is it connected, giving the impression that the IP:PORT provided by the user was successful even when it has not been.

**Comment by rustyrussell** [comment]
Yes, it will try the hint, then fallback to other sources.  We should accurately report the address which *worked*, IMHO.

**Comment by dscotese** [comment]
Simply because I see a principle here that I believe is difficult to recognize, I wanted to highlight my "misuse."  I recognized as misuse while I was doing it and so no problems resulted.  A user "flying blind" may try to remember the correct IP of his node and discover that his first guess was correct!  OR so it seems.  That could create an extremely confusing and unfortunate situation.

The principle is that misuse, as I executed it, can be intentional, but can also be unintentional, and so whenever software solves a problem the user might not know about, there ought to be an indicator that there was a problem that it solved.  In this case, __"You provided an IP that doesn't work.  This one _____ does work."__  If not to make a feature or function more useful, at least to avoid misleading those who don't understand the sophistication of the software (me).  Human relations have the same feature, so if something someone said sounds like nonsense and you figure out what they meant, let them know what you (think) they actually meant, at least so they know a better way to say it. :-)

P.S. I was inspired to write this comment by Eric von Hippel's book, Democratizing Innovation.
